### PR TITLE
Add Spring Security and allow unrestricted access to /info.json and /…

### PR DESCRIPTION
**Pull request by wtritch**
…health.json endpoints as well as all non-actuator endpoints.

### CODE REVIEW

File: common/pom.xml [hunk 1]
```diff
@@ -45,22 +58,59 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
+            <version>${spring.boot.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+            <version>${spring.boot.version}</version>
+        </dependency>
```
Comment (wtritch) [inline common/pom.xml:93]: Small re-organization in here, this spring-boot-starter-security is the only new dependency.

File: common/pom.xml [hunk 2]
```diff
@@ -37,6 +38,18 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
```
Comment (millerts) [inline common/pom.xml:41]: Why is this needed?

**Review by millerts — APPROVED** [review]

File: common/src/main/java/tds/common/configuration/SecurityConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(final HttpSecurity http) throws Exception {
+        http
+            .authorizeRequests()
+            .antMatchers("/info*", "/health*").permitAll();
+    }
+
+}
```
Comment (wtritch) [inline common/src/main/java/tds/common/configuration/SecurityConfiguration.java:27]: This security configurer leaves all endpoints unprotected (as before) except for the Spring Actuator endpoints, which are now protected.  See ManagementWebSecurityAutoConfiguration if you want to dive into the spring magick.

File: common/pom.xml (hunk 2 above)
Comment (wtritch) [reply common/pom.xml:41]: oh, shoot,  this is the other new dependency.  This is needed by the spring-security WebSecurityConfigurerAdapter.  Think I should switch it to "provided" ?

Comment (millerts) [reply common/pom.xml:41]: I think we should try it.

Comment (wtritch) [reply common/pom.xml:41]: Specifically, there's a dependency on "javax.servlet.Filter"

File: common/pom.xml (hunk 1 above)
Comment (jtreuting) [inline common/pom.xml:93]: This block was just moved below right?

**Review by jtreuting — APPROVED** [review]

File: common/pom.xml (hunk 2 above)
Comment (wtritch) [reply common/pom.xml:41]: yeah, "provided" works fine for our microservices, since they're all including some form of servlet container.

File: common/pom.xml (hunk 1 above)
Comment (wtritch) [reply common/pom.xml:93]: right, I put the spring stuff together, broke out the spring.boot.version into a property, moved some non-test dependencies out of the \ section, and added a \<scope>test\</scope> to one of the actual test dependencies.
